### PR TITLE
To do list add & remove requirements

### PR DESCRIPTION
**Pull request by d4nQw3rty**
### General requirements:

- Make sure there are [no linter errors](https://github.com/microverseinc/linters-config).

- Make sure to use a correct [GitHub Flow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/github_flow.md).

- Make sure the work is documented [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).

- Follow Microverse's list of the [best practices for HTML & CSS](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md).

- Follow Microverse's list of [best practices for JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).

### Project requirements:

- Remove all hardcoded items from the tasks array.
- Create a new JavaScript file for the new functionality.
- Implement a function for adding a new task (add a new element to the array).
- Implement a function for adding a new task (add a new element to the array).
- Implement a function for editing task descriptions.
- By default new tasks should have the property completed set to false and the property index set to the value of the new array length.
- Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique(i.e. if you're deleting the first task index 1 from the list, the index of the next task(2) should be set to 1).
- All changes to the To Do List should be saved in local storage.

### CODE REVIEW

File: src/modules/taskConstructor.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+export default class Task {
+  constructor(description, id = 0) {
+    this.description = description;
+    this.editable = false;
+    this.completed = false;
+    this.id = id;
+  }
+}
```
Comment (ShahierNashaat) [inline src/modules/taskConstructor.js]: - [x] Kindly notice that there is no need for the `editable` attribute, also the `id` attribute should be the `index` attribute.

```javascript
export default class Task {
  constructor(description, index = 0) {
    this.description = description;
    this.completed = false;
    this.index = index;
  }
}
```

File: src/modules/storageMethods.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+      localStorage.setItem('tasks', JSON.stringify(tasks));
+    }
+    return tasks;
+  }
+
+  static add(description) {
+    if (!description) return -1;
+    const storage = localStorage.getItem('tasks');
+    const tasks = storage ? JSON.parse(storage) : [];
+    const task = new Task(description, tasks.length);
```
Comment (ShahierNashaat) [inline src/modules/storageMethods.js]: - [x] Kindly notice that if a 5th task is added to the list, the index of that task should equal to 5, so please fix this by adding one to the task.length.

```javascript
const task = new Task(description, tasks.length + 1);
```

**Review by ShahierNashaat — CHANGES_REQUESTED** [review]
Hi @d4nQw3rty, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 💯 

- Correct use of GitFlow ✔️
- Descriptive commit messages ✔️
- Professional looking pull request ✔️
- Your readme looks professional ✔️
- Good overall design ✔️
- Your code is clean and well documented ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by zainsadaqat — APPROVED** [review]
# APPROVED!!!! 🎉 🎉 🎉 🎉 🎉 🎉
## Hey @d4nQw3rty  😊 😃 

#### Great Job 😄 👏 You've done it. Your project is complete!  There is nothing else to say other than... it's time to merge it 👍🏻 ✅ 

# Congratulations!!! 🎉 🥳 👏 🙌 🎊 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**